OpenRA.Mods.D2k/Traits/Sandworm.cs

```diff
@@ -111,6 +111,10 @@ void RescanForTargets(Actor self)
 
 			while (!self.World.Map.Contains(moveTo) || !mobile.CanEnterCell(moveTo, null, false))
 			{
+				// without this check, this while statement can be infinity loop
```

Comment:
>  move the (moveTo == self.Location)

not moved, but copied this check into while statement and leave additional check after while statement.